Gordon360/Services/MembershipService.cs

```diff
@@ -19,14 +21,70 @@ namespace Gordon360.Services
     public class MembershipService : IMembershipService
     {
         private readonly CCTContext _context;
-        private IAccountService _accountService;
+        private readonly IAccountService _accountService;
 
         public MembershipService(CCTContext context, IAccountService accountService)
         {
             _context = context;
             _accountService = accountService;
         }
 
+        /// <summary>
+        /// Fetches the memberships associated with the activity whose code is specified by the parameter.
+        /// </summary>
+        /// <param name="activityCode">Optional activity code filter</param>
+        /// <param name="username">Optional username filter</param>
+        /// <param name="sessionCode">Optional session code, defaults to current session. Use "*" for all sessions</param>
+        /// <param name="participationTypes">Optional filter for involvement participation types (MEMBR, ADV, LEAD, GUEST, GRP_ADMIN)</param>
+        /// <returns>An IEnumerable of the matching MembershipView objects</returns>
+        public IEnumerable<MembershipView> GetMemberships(
+            string? activityCode = null,
+            string? username = null,
+            string? sessionCode = null,
+            List<string>? participationTypes = null
```

Comment:
I would think we want to validate our parameters as early on as possible.

Reply:
It doesn't take the enum type because we don't compare against the enum type (since we have to do a string comparison for it to be convertible to SQL).

We COULD have the controller explicitly convert the string to the enum, and then have the service convert the enum back to the string for comparison with the database string. However, given that that wouldn't actually change the behavior at all (unless we want to throw a BadRequest response when the client sends invalid participation types), that seems like unnecessary overhead to me.